metadata/src/main/java/org/apache/kafka/controller/metrics/ControllerMetadataMetricsPublisher.java

```diff
@@ -115,6 +115,9 @@ private void publishDelta(MetadataDelta delta) {
             }
         }
         changes.apply(metrics);
+        if (delta.featuresDelta() != null) {
```

Comment:
I wonder if we should only set the zk state metric if we're on a metadata.version that supports it. Otherwise we could see clusters go from NONE -> MIGRATION -> POST_MIGRATION, which is a bit odd. (NONE is supposed to be a terminal state)

Reply:
That's fine although it could also be a future improvement

Reply:
https://issues.apache.org/jira/browse/KAFKA-14939